ibm_cloud_sdk_core/authenticators/authenticator.py

```diff
@@ -0,0 +1,46 @@
+# coding: utf-8
+
+# Copyright 2019 IBM All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+
+class Authenticator(ABC):
+    @abstractmethod
+    def authenticate(self):
+        """
+        Returns the (username, password) or bearer <token>
+        """
+        pass
+
+    @abstractmethod
+    def validate(self):
+        """
+        Checks if all the inputs needed are present
+        """
+        pass
+
+    @abstractmethod
+    def _is_basic_authentication(self):
+        """
+        Return true if basic authentication
+        """
+        pass
+
+    @abstractmethod
+    def _is_bearer_authentication(self):
+        """
+        Return true if bearer authentication
+        """
+        pass
```

Comment:
What are these methods `_is_bearer_authentication` `_is_basic_authentication` used for? I don't know that we necessarily need them